zyte_common_items/processors.py

```diff
@@ -129,3 +131,87 @@ def simple_price_processor(value: Union[Selector, HtmlElement], page: Any) -> An
     """
     price = extract_price(value)
     return _format_price(price)
+
+
+@only_handle_nodes
+def description_html_processor(value: Union[Selector, HtmlElement], page: Any) -> Any:
+    """Convert the data into a cleaned up HTML if possible.
+
+    Uses the clear-html_ library.
+
+    Supported inputs are :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    Other inputs are returned as is.
+
+    Puts the cleaned HtmlElement object into ``page._descriptionHtml_node``.
+
+    .. _clear-html: https://github.com/zytedata/clear-html
+    """
+    if isinstance(value, Selector):
+        value = value.root
+    if value is None:
+        return None
+    assert isinstance(value, HtmlElement)
+    cleaned_node = clean_node(value, _get_base_url(page))
+    page._descriptionHtml_node = cleaned_node
+    return cleaned_node_to_html(cleaned_node)
+
+
+def description_processor(value: Any, page: Any) -> Any:
+    """Convert the data into a cleaned up text if possible.
+
+    Uses the clear-html_ library.
+
+    Supported inputs are :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    Other inputs are returned as is.
+
+    Puts the cleaned HtmlElement object into ``page._description_node`` and the
+    cleaned text into ``page._description_str``.
+
+    .. _clear-html: https://github.com/zytedata/clear-html
+    """
+    value = _handle_selectorlist(value)
+    if isinstance(value, str):
+        page._description_str = value
+        return value
+    if isinstance(value, Selector):
+        value = value.root
+    if value is None:
+        return None
+    assert isinstance(value, HtmlElement)
+    cleaned_node = clean_node(value, _get_base_url(page))
+    cleaned_text = cleaned_node_to_text(cleaned_node)
+    page._description_node = cleaned_node
+    page._description_str = cleaned_text
+    return cleaned_text
+
+
+def wrap_description_into_html(description: str) -> str:
```

Comment:
What do you think about making this function private for now? We might need to change the name later, or change the implementation, because a similar function could be used for other similar fields in other data types, which may not be "description".